lib/check_regional_records.rb

```diff
@@ -126,6 +126,60 @@ def self.load_ordered_results(event_id, competition_id, value_column, regional_r
+    SOLUTION_TYPES.to_h do |value_column, value_name|
+      if competition_id.present?
+        competition = Competition.find(competition_id)
+        results = competition.results
+        results = competition.results.where(event_id: event_id) if event_id.present?
+        minimum_results_by_event = results.select("id, event_id, competition_id, round_type_id, country_id, MIN(#{value_column}) AS `value`")
+                                          .where.not(value_column => ..0)
+                                          .group("id, event_id, competition_id, round_type_id, country_id")
+        timestamps = ResultTimestamp.where(result_id: minimum_results_by_event.ids)
+        candidates = minimum_results_by_event.filter_map do |result|
+          timestamp = timestamps.find { |t| t.result_id == result.id }.round_timestamp
+          record_at_time_of_round = RegionalRecord.record_for(result.event_id, value_name, :NR, country_id: result.country_id, date: timestamp)
+          [result.event_id, result.competition_id, result.round_type_id, result.country_id, timestamp, result.value] if record_at_time_of_round > result.value
+        end
+        [value_column, candidates]
+      else
+        # Currently takes (42717.7 ms) for all results (16080.0 ms) when filtering for event_id
+        best_at_date = ActiveRecord::Base.connection.execute(<<~SQL)
+                                                  WITH ranked_results AS (
+                                                  SELECT
+                                                    results.id AS result_id,
+                                                    results.event_id,
+                                                    results.country_id,
+                                                    results.#{value_column},
+                                                    result_timestamps.round_timestamp,
+                                                    ROW_NUMBER() OVER (
+                                                      PARTITION BY results.event_id, result_timestamps.round_timestamp
+                                                      ORDER BY results.#{value_column}
+                                                    ) AS best_rank
+                                                  FROM results
+                                                  INNER JOIN result_timestamps ON result_timestamps.result_id = results.id
+                                                  WHERE results.#{value_column} > 0
+                                                  #{event_id.present? ? "AND results.event_id = '#{event_id}'" : ''}
+                                                  #{from_timestamp.present? ? "AND result_timestamps.round_timestamp >= '#{from_timestamp}'" : ''}
+                                                  #{to_timestamp.present? ? "AND result_timestamps.round_timestamp <= '#{to_timestamp}'" : ''}
+                                                )
```

Comment:
I think you will be able to push runtimes down by a lot if you pre-merge all of this information into `result_timestamp`.

This will effectively mean that the code you're running here is not 100% live (it is only as recent as the state of the precomputed `results_timestamp` table is) but (a) WRT is already used to that with the existing `regional_records_lookup` and (b) the potential performance gains are worth it IMO.